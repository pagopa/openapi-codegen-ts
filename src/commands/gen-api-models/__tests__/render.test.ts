/* eslint-disable sonarjs/no-duplicate-string */

import { OpenAPIV2 } from "openapi-types";
import * as SwaggerParser from "swagger-parser";

import {
 renderDefinitionCode,
} from "../render";

let spec: OpenAPIV2.Document;
beforeAll(
  async () =>
    (spec = (await SwaggerParser.bundle(
      `${process.cwd()}/__mocks__/api.yaml`
    )) as OpenAPIV2.Document)
);

describe("renderDefinitionCode", () => {
  
    it.each`
      case   | definitionName
      ${"1"} | ${"Message"}
      ${"2"} | ${"DefinitionFieldWithDash"}
    `("should render $case", async ({ definitionName }) => {
      if (!spec.definitions) {
        fail("dadsasa");
      }
      const code = await renderDefinitionCode(
        definitionName,
        spec.definitions[definitionName],
        true,
        false
      );
      expect(code).toMatchSnapshot();
    });

});
