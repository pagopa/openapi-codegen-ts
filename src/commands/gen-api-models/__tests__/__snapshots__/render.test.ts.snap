// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderDefinitionCode should render 1 1`] = `
"/**
 * Do not edit this file it is auto-generated by io-utils / gen-api-models.
 * See https://github.com/pagopa/io-utils
 */
/* eslint-disable  */

import { MessageContent } from \\"./MessageContent\\";
import * as t from \\"io-ts\\";

// required attributes
const MessageR = t.interface({
  id: t.string,

  content: MessageContent
});

// optional attributes
const MessageO = t.partial({
  sender_service_id: t.string
});

export const Message = t.exact(t.intersection([MessageR, MessageO], \\"Message\\"));

export type Message = t.TypeOf<typeof Message>;
"
`;

exports[`renderDefinitionCode should render 2 1`] = `
"/**
 * Do not edit this file it is auto-generated by io-utils / gen-api-models.
 * See https://github.com/pagopa/io-utils
 */
/* eslint-disable  */

import * as t from \\"io-ts\\";

// required attributes
const DefinitionFieldWithDashR = t.interface({});

// optional attributes
const DefinitionFieldWithDashO = t.partial({
  \\"id-field\\": t.string
});

export const DefinitionFieldWithDash = t.exact(
  t.intersection(
    [DefinitionFieldWithDashR, DefinitionFieldWithDashO],
    \\"DefinitionFieldWithDash\\"
  )
);

export type DefinitionFieldWithDash = t.TypeOf<typeof DefinitionFieldWithDash>;
"
`;
